caste.go

```diff
@@ -990,6 +990,188 @@ func ToStringMapE(i interface{}) (map[string]interface{}, error) {
 	}
 }
 
+// ToStringMapIntE casts an interface to a map[string]int{} type.
+func ToStringMapIntE(i interface{}) (map[string]int, error)  {
+	var m = map[string]int{}
+	if i == nil {
+		return m, fmt.Errorf("unable to cast %#v of type %T to map[string]int", i, i)
+	}
+
+	switch v := i.(type) {
+	case map[interface{}]interface{}:
+		for k, val := range v {
+			m[ToString(k)] = ToInt(val)
+		}
+		return m, nil
+	case map[string]interface{}:
+		for k, val := range v {
+			m[k] = ToInt(val)
+		}
+		return m, nil
+	case map[string]int:
+		return v, nil
+	case string:
+		err := jsonStringToObject(v, &m)
+		return m, err
+	}
+
+	if reflect.TypeOf(i).Kind() != reflect.Map {
+		return m, fmt.Errorf("unable to cast %#v of type %T to map[string]int", i, i)
+	}
+
+	kind := reflect.TypeOf(i).Elem().Kind()
+	switch kind {
```

Comment:
Since we're using reflection here, this should be possible to make much shorter. I suggest you just range over the map using map and use `ToIntE` and `SetMapIndex`.

Reply:
Cool, I didn't know you can do that. Do you mean something like this?

```
mVal := reflect.ValueOf(m)
v := reflect.ValueOf(i)
for _, keyVal := range v.MapKeys() {
	mVal.SetMapIndex(keyVal, reflect.ValueOf(ToInt64(v.MapIndex(keyVal).Interface())))
}
return m, nil
```

Reply:
Yes, something like that, but you should probably use ToInt64E to improve error handling.

Reply:
I've updated the PR